cli/src/main/java/org/aya/cli/repl/ReplCommands.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2020-2021 Yinsen (Tesla) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.cli.repl;
+
+import kala.collection.immutable.ImmutableSeq;
+import org.aya.api.util.NormalizeMode;
+import org.aya.cli.repl.command.Command;
+import org.aya.pretty.doc.Doc;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public interface ReplCommands {
```

Comment:
Do you think it makes sense to make a constructor for Command that takes a lambda?

Reply:
It is impossible. The `@Entry` method can take arbitrary arguments and we will provide completion by checking their types. Lambdas can only have fixed arguments? Are there any container that supports storing vararg (not `Object`) lambdas?

Reply:
I see, so it's based on annotation processing!!

Reply:
Yeah!